### PR TITLE
Fix format.py to use jit-format.exe, not jit-format.bat

### LINKED ISSUE #61331: Azure Pipelines / runtime (Formatting windows x64) fails
Azure Pipelines / runtime (Formatting windows x64) leg fails in https://github.com/dotnet/runtime/pull/61320:
```log
The process 'C:\hostedtoolcache\windows\Python\3.10.0\x64\python.exe' failed with exit code 1

Publishing build artifacts failed with an error: Not found PathtoPublish: D:\a\1\s\src\coreclr\format.patch
```
Error details:
```log
File "C:\hostedtoolcache\windows\Python\3.10.0\x64\lib\subprocess.py", line 1435, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
FileNotFoundError: [WinError 2] The system cannot find the file specified
##[error]The process 'C:\hostedtoolcache\windows\Python\3.10.0\x64\python.exe' failed with exit code 1
```
see more: https://dev.azure.com/dnceng/public/_build/results?buildId=1459308&view=logs&j=35546789-40c1-5b36-592c-13cb6b78e0dd&t=ee62566f-6723-5502-9bef-329d30faf067&l=115

### PR DESCRIPTION
**Pull request by BruceForstall**
Adjust for https://github.com/dotnet/jitutils/pull/343

Fixes https://github.com/dotnet/runtime/issues/61331

### CODE REVIEW

**Comment by BruceForstall** [comment]
@hoyosjs @AndyAyersMS PTAL
cc @dotnet/jit-contrib 

This shouldn't wait for PR testing, since it fixes the CI and doesn't affect anything except the jit-format job.

**Review by hoyosjs — APPROVED** [review]

**Comment by hoyosjs** [comment]
Merging to unblock PRs touching JIT sources.

**Pull request merged**
